lsp-test/src/Language/LSP/Test.hs

```diff
@@ -158,9 +159,19 @@ runSessionWithConfig :: SessionConfig -- ^ Configuration options for the session
                      -> FilePath -- ^ The filepath to the root directory for the session.
                      -> Session a -- ^ The session to run.
                      -> IO a
-runSessionWithConfig config' serverExe caps rootDir session = do
+runSessionWithConfig = runSessionWithConfig' id
+
+-- | Starts a new session with a custom configuration.
+runSessionWithConfig' :: (CreateProcess -> CreateProcess)
```

Comment:
Can we give this a more descriptive name? `runSessionInCustomProcess` or something?

I also wonder if it would make more sense to just take a user-specified `CreateProcess` rather than taking a modification function? We could also then remove e.g. the `serverExe` argument, which is somewhat clean. I'm not sure what the nicest way to factor this is.

Reply:
Sure, happy to tweak the name once we're settled on the design! Responded to the other suggestion below